Detectors/TOF/reconstruction/src/Encoder.cxx

```diff
@@ -0,0 +1,325 @@
+#include "TOFReconstruction/Encoder.h"
+#include "TOFReconstruction/Decoder.h"
+#include <iostream>
+#include <chrono>
+#include <algorithm>
+#include "CommonConstants/LHCConstants.h"
+#define VERBOSE
+
+namespace o2
+{
+namespace tof
+{
+namespace compressed
+{
+
+bool Encoder::open(std::string name)
+{
+  if (mFile.is_open()) {
+    std::cout << "Warning: a file was already open, closing" << std::endl;
+    mFile.close();
+  }
+  mFile.open(name.c_str(), std::fstream::out | std::fstream::binary);
+  if (!mFile.is_open()) {
+    std::cerr << "Cannot open " << name << std::endl;
+    return true;
+  }
+  return false;
+}
+
+bool Encoder::flush()
+{
+  mFile.write(mBuffer, mIntegratedBytes);
+  return false;
+}
+
+bool Encoder::close()
+{
+  if (mFile.is_open())
+    mFile.close();
+  return false;
+}
+
+bool Encoder::alloc(long size)
+{
+  mSize = size;
+  mBuffer = new char[mSize];
```

Comment:
same comment about the mBuffer here (avoid new)

Reply:
It is not clear to me how to use a vector.
mBuffer is used to cache memory writing byte by byte and then to flush to a file.
Does it work if memory is allocated as a vector?
Regarding unique_pointer. I have never used such a feature. Can you point me to an example?

Reply:
If you want to have explicitly mBuffer pointer (e.g. to decode from an external memory block in future), you can add std::vector<char> mBufferLocal data member and subtitute the lines above by 
````
mBufferLocal.resize(size);
mBuffer = mBufferLocal.data();
````

For the unique_ptr option, you can have data member
``std::unique_ptr<char[]> mBufferLocal;``
and initialize it with 
``mBufferLocal = std::make_unique<char>(size);``
You use it as a usual pointer, with ``mBufferLocal[i]``, and you can convert extract its raw pointer via ``mBufferLocal.get()``.
You don't need to clean it (but can do this with ``mBufferLocal.release()``), it will be done automatically on deletion of the object.

Reply:
Ok. I got the point. Thanks.
I implemented the vector solution